src/meshes/p4est_mesh.jl

```diff
@@ -469,12 +474,186 @@ function p4est_mesh_from_standard_abaqus(meshfile, mapping, polydeg,
 
     p4est = new_p4est(connectivity, initial_refinement_level)
 
-    # There's no simple and generic way to distinguish boundaries. Name all of them :all.
-    boundary_names = fill(:all, 2 * n_dimensions, n_trees)
+    if boundary_symbols === nothing
+        # There's no simple and generic way to distinguish boundaries. Name all of them :all.
+        boundary_names = fill(:all, 2 * n_dimensions, n_trees)
+    else # Boundary information supplied
+        # Read in nodes belonging to boundaries
+        node_set_dict = parse_node_sets(meshfile, boundary_symbols)
+        # Read in all elements with associated nodes to specify the boundaries
+        element_node_matrix = parse_elements(meshfile, n_trees, n_dimensions)
+
+        # Initialize boundary information matrix with symbol for no boundary / internal connection
+        boundary_names = fill(Symbol("---"), 2 * n_dimensions, n_trees)
+
+        if n_dimensions == 2
+            for tree in 1:n_trees
+                tree_nodes = element_node_matrix[tree, :]
+                # For node labeling, see 
+                # https://docs.software.vt.edu/abaqusv2022/English/SIMACAEELMRefMap/simaelm-r-2delem.htm#simaelm-r-2delem-t-nodedef1
+                # and search for "Node ordering and face numbering on elements"
+                for boundary in keys(node_set_dict)
+                    # Check bottom edge
+                    if tree_nodes[1] in node_set_dict[boundary] &&
+                       tree_nodes[2] in node_set_dict[boundary]
+                        # Bottom boundary is position 3 in p4est indexing
+                        boundary_names[3, tree] = boundary
+                    end
+                    # Check right edge
+                    if tree_nodes[2] in node_set_dict[boundary] &&
+                       tree_nodes[3] in node_set_dict[boundary]
+                        # Right boundary is position 2 in p4est indexing
+                        boundary_names[2, tree] = boundary
+                    end
+                    # Check top edge
+                    if tree_nodes[3] in node_set_dict[boundary] &&
+                       tree_nodes[4] in node_set_dict[boundary]
+                        # Top boundary is position 4 in p4est indexing
+                        boundary_names[4, tree] = boundary
+                    end
+                    # Check left edge
+                    if tree_nodes[4] in node_set_dict[boundary] &&
+                       tree_nodes[1] in node_set_dict[boundary]
+                        # Left boundary is position 1 in p4est indexing
+                        boundary_names[1, tree] = boundary
+                    end
+                end
+            end
+        else # n_dimensions == 3
+            # Admitted 3D element: C3D8
```

Comment:
Would it be worthwhile to write a helper function like `assign_boundary_name` or something that dispatches on the ABAQUS element type? It might make this constructor routine a bit easier to read. Also, it might grant us the flexibility to reuse your strategy to assign boundary names for the `T8codeMesh` type.

Reply:
Yes, that sounds reasonable to me.